packages/block-library/src/site-title/edit.js

```diff
@@ -1,6 +1,8 @@
 /**
  * WordPress dependencies
  */
+import { compose } from '@wordpress/compose';
+import { withSelect } from '@wordpress/data';
```

Comment:
`withSelect` uses [`useSelect`](https://developer.wordpress.org/block-editor/packages/packages-data/#useSelect) under the hood and we are moving towards using that to better compose with other hooks, take advantage of functional component performance improvements, and improve readability.